src/transformers/models/whisper/convert_openai_to_hf.py

```diff
@@ -51,6 +60,20 @@
 }
 
 
+def _get_generation_config(is_multilingual: bool, num_languages: int = 100) -> GenerationConfig:
```

Comment:
Thanks for adding this! The only generation config attribute that is checkpoint specific is the alignment heads: https://gist.github.com/hollance/42e32852f24243b748ae6bc1f985b13a

The alignment can only really be worked out by looking at the cross-attention plots: https://github.com/openai/whisper/blob/main/notebooks/Multilingual_ASR.ipynb

=> since it's checkpoint specific, I think we should remove this attribute from the generation config. The user will then be prompted to set it themselves if they require word-level timestamps: https://github.com/huggingface/transformers/blob/ef716736165cbbac433a3c4b2c3ea97246b3d6b9/src/transformers/models/whisper/modeling_whisper.py#L1973

WDYT @flyingleafe @ArthurZucker?

```python
if hasattr(generation_config, "alignment_heads"):
    delattr(generation_config, "alignment_heads")
```

Reply:
Agreed 😉

Reply:
@sanchit-gandhi The alignment heads can be copy-pasted right from the [OpenAI repository](https://github.com/openai/whisper/blob/fcfeaf1b61994c071bba62da47d7846933576ac9/whisper/__init__.py#L34), without looking at the cross-attention plots. They are provided there in quite a compact way.

Let us set up the alignment heads directly from this dictionary if the user provided the version of OpenAI model, and skip setting those (with a warning to the user) if the checkpoint is custom.

Reply:
Sure! If you have a clean way of determining whether the checkpoint is 'official' or 'custom' this works!